command/git.go

```diff
@@ -193,9 +196,16 @@ func (c *gitFetchProject) ParseParams(params map[string]interface{}) error {
 	return nil
 }
 
+func (c *gitFetchProject) gitResetWithUnshallow(revision string) []string {
+	return []string{
+		fmt.Sprintf("git reset --hard %s && exit 0", revision),
```

Comment:
Doesn't this make everything below unreachable?

Reply:
It makes the rest unreachable only if git reset succeeds.